File: tree/isXXTree/isFullTree.java
//0 or 2 children
public boolean isFull(TreeNode root) {
        // if empty tree
    if(node == null)
        return true;

        // if leaf node
        if(node.left == null && node.right == null )
            return true;

        // if both left and right subtrees are not null
        // the are full
        if((node.left!=null) && (node.right!=null))
            return isFullTree(node.left) && isFullTree(node.right);

        // if none work
        return false;
    }
  }

File: tree/isXXTree/isFullTree.py
# except leaf node, other nodes should have 2 nodes
def isfulltree(root):
    # edge case
    if root is None:
        return True
    # base case: if leaf node:
    if root.left is None and root.right is None:
        return True
    # other nodes
    if root.left and root.right:
        return isfulltree(root.left) and isfulltree(root.right)
    return False
